mllib-tests/v1p5/src/main/scala/mllib/perf/MLAlgorithmTests.scala

```diff
@@ -214,18 +237,31 @@ class GLMClassificationTest(sc: SparkContext) extends GLMTests(sc) {
     val loss = stringOptionValue(LOSS)
     val regType = stringOptionValue(REG_TYPE)
     val regParam = doubleOptionValue(REG_PARAM)
+    val elasticNetParam = doubleOptionValue(ELASTIC_NET_PARAM)
     val numIterations = intOptionValue(NUM_ITERATIONS)
     val optimizer = stringOptionValue(OPTIMIZER)
 
+    // For classification problem in GLM, we currently support logistic loss and hinge loss.
     if (!Array("logistic", "hinge").contains(loss)) {
       throw new IllegalArgumentException(
         s"GLMClassificationTest run with unknown loss ($loss).  Supported values: logistic, hinge.")
     }
-    if (!Array("none", "l1", "l2").contains(regType)) {
-      throw new IllegalArgumentException(s"GLMClassificationTest run with unknown regType" +
-        s" ($regType).  Supported values: none, l1, l2.")
-    }
-    if (!Array("sgd", "lbfgs").contains(optimizer)) {
+
+    if (Array("sgd").contains(optimizer)) {
+      if (!Array("none", "l1", "l2").contains(regType)) {
+        throw new IllegalArgumentException(
+          s"GLMRegressionTest run with unknown regType ($regType) with sgd.  Supported values: none, l1, l2.")
+      }
+    } else if (Array("lbfgs").contains(optimizer)) {
+      if (Array("hinge").contains(loss)) {
```

Comment:
I would check `!Array("logistic")` instead to guard against `optimizer=lbfgs, loss=some_nonsense`